### PR TITLE
chore(ml): updated dockerfile, added typing, packaging

### PR DESCRIPTION
**Pull request by mertalev**
This PR does several things:
- Added packaging using Poetry. This codifies the current ML dependencies with `pyproject.toml` and `poetry.lock` for reproducible and faster builds.
- Added typing for FastAPI endpoints. This allows for runtime validation of requests and responses and codifies the ML server's API.
  - I actually wrote an OpenAPI spec for this and used it to partially generate the `schemas.py` file with [datamodel-code-generator](https://pypi.org/project/datamodel-code-generator/). However, I didn't include it since I'm unsure if it's a useful addition at the moment without deeper integration. On the one hand, Pydantic data models are already a pythonic approach to API modeling without the need for an explicit spec. On the other hand, this could be used to integrate the ML server's API with remote `$ref`'s in the future.
- Updated the Dockerfile to benefit from the `poetry.lock` file.
- Updated the README to reflect the addition of Poetry.

The actual behavior of the server once launched should be unchanged.

### CODE REVIEW

File: machine-learning/pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,37 @@
+[tool.poetry]
+name = "machine-learning"
+version = "1.59.1"
```
Comment (jrasm91) [inline machine-learning/pyproject.toml:3]: Is this something we'll need to bump each release? If so can you take a look at the pump script?

**Review by jrasm91 — APPROVED** [review]

File: machine-learning/pyproject.toml (hunk 1 above)
Comment (mertalev) [reply machine-learning/pyproject.toml]: Ooh, I didn't notice that script. It should work for this file too now but only if it's used from the project root.

File: machine-learning/app/schemas.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+class TextResponse(BaseModel):
+    __root__: str
+
+
+class MessageResponse(BaseModel):
+    message: str
+
+
+class TagResponse(BaseModel):
+    __root__: list[str]
```
Comment (alextran1502) [inline machine-learning/app/schemas.py:33]: I assume `__root__` here mean the `TagReponse` model itself is considered as a list of `str`, correct?

**Review by alextran1502 — CHANGES_REQUESTED** [review]
I ran into this error in development.

image

Have you tested this change in production build?

**Comment by mertalev** [comment]
Hmm, I only tested with the production docker compose. The dev version needs `../machine-learning/src` to be changed to `../machine-learning/app`.

File: machine-learning/app/schemas.py (hunk 2 above)
Comment (mertalev) [reply machine-learning/app/schemas.py:33]: Yup, it's basically a type alias.

**Review by alextran1502 — APPROVED** [review]

**Pull request merged**